library/src/androidMain/kotlin/org/eyeseetea/dhis2/lightsdk/common/ReadFileJVM.kt

```diff
@@ -0,0 +1,20 @@
+package org.eyeseetea.dhis2.lightsdk.common
+
+import java.io.File
+import java.io.InputStream
+
+internal actual fun readFile(file: String): String {
```

Comment:
I think that we should intent does not pass the path by parameter, only the file name. The readfile fun should find out the path.
I think we should use classLoader as in FileReader in malariaapp

Reply:
@xurxodev I could not move these files to commonTest.
The file ReadFileJVM is on androidMain package.
The file ReadFileJS is on jsMain package.
And the ReadFile is in commonMain.

When I tried to move (ReadFileJVM or ReadFileJS) the file don't compile.
The specific js or java code can't be placed into the main package, and the IDE shows the error.

And when I tried to move the interface(ReadFile.kt) to the test package, the specific files the IDE show a declaration error because they don't have access to it from the common code.

Reply:
@idelcano for JVM the solution in pictureapp and malaria app working, I have dedicated a time to test with this class working and I'ts better than have hardcoded paths:
actual class FileReader {
    actual fun readFile(file: String): String {
        logDebug("!!!!The file to read:$file")

        val inputStream: InputStream = file(javaClass, file).inputStream()
        val inputString = inputStream.bufferedReader().use { it.readText() }
        return inputString
    }

    private fun file(clazz: Class<*>, filename: String): File {
        val classLoader = clazz.classLoader
        val resource = classLoader.getResource(filename)

        logDebug("!!!!The path file to read:${resource.path}")
        return File(resource!!.path)
    }
}